concepts/regular-expressions/about.md

```diff
@@ -0,0 +1,160 @@
+const myArray = str.match(re);
+console.log(myArray);
+
+=> // ["Home", "home"]
+```
+
+```javascript
+let regex = /[a-z]/gi; // literal notation
+let regex = new RegExp('[a-z]', 'gi'); // constructor with string pattern as first argument
+let regex = new RegExp(/[a-z]/, 'gi'); // constructor with regular expression literal as first argument (Starting with ECMAScript 6)
+```
+
+## 4. Most common Functions
+
+When regular expressions are combined with current build-in functions in JavaScript, we have some powerful ways of manipulating and getting data from strings.
+
+These are some of the most common functions used alongside regex.
+
+### Test
+
+```javascript
+const str = 'It is difficult to test if you have a virus';
+const result = /virus$/.test(str);
+
+console.log(result); // true
+```
+
+### Match
+
+In this way, we are able both to **search** and to **extract** information from any string. For example:
+
+```javascript
+const funnyQuote =
+  'If you see someone crying, ask if it is because of their haircut.';
+const regex1 = /someone/;
+const regex2 = /happy/;
+
+funnyQuote.match(regex1); // ["someone", index: 3, input: "If you see someone crying, ask if it is because of their haircut.", groups: undefined]
+
+funnyQuote.match(regex2); // null
+```
+
+When the Global Search flag `/g` is present, instead of getting the only match alongside useful information such as the index or input, the method returns all of the occurances displayed in the array:
+
+```javascript
+const funnyQuote =
+  'If you see someone crying, ask if it is because of their haircut.';
+
+const regex3 = /if/gi;
+
+funnyQuote.match(regex3); // ["If", "if"];
+```
+
+### Replace
+
+The [replace()][mdn-regex-replace] method in JavaScript allows us to search for a value within a given string, and replacing it with a desired new value.
+
+```javascript
+string.replace(searchValue, newValue);
+```
+
+The pattern for searching the substitution can be a single string, or a regular expression.
+
+```javascript
+let string = 'I like dogs!';
+let result = string.replace('dogs', 'cats');
+
+let string = 'I would love to travel to Japan';
+let result = string.replace(/Japan/g, 'Hawaii');
+```
+
+Moreover, we can apply a function on the replacement position in order to make additional changes to each value.
+
+```javascript
+let text = 'Say hello to the chatbot.';
+let result = text.replace(/chatbot|hello/gi, function (word) {
+  return word.toUpperCase();
+});
+// -> "Say HELLO to the CHATBOT"
+```
+
+### Split
+
+The [split()][mdn-regex-split] method in JavaScript represents a different way of using and manipulating strings with regular expressions.
+
+```javascript
+const str = 'hello,user.how are.you';
+
+const result = str.split(/[,.\s]/);
+
+console.log(result);
+//--> ['hello', 'user', 'how', 'are', 'you']
+```
+
+## 5. Performance with Regex
+
+Regarding performance, both of them create a RegExp object. The main difference is how often the regex is compiled:
```

Comment:
The reference for "both of them" is not clear here. To fix this and since you cover more than performance here, maybe it would be better to make this whole section part of "creating regex" above.